en/lessons/advanced/otp-distribution.md

```diff
@@ -0,0 +1,430 @@
+* The function to be executed by the remote process running on that node
+
+It establishes the connection to the remote node and executes the given function on that node, returning the PID of the linked process.
+
+Let's define a module, `Kate`, in the `kate` node that knows how to introduce Kate, the person:
+
+```elixir
+iex(kate@localhost)2> defmodule Kate do
+...(kate@localhost)2>   def say_name do
+...(kate@localhost)2>     IO.puts "Hi, my name is Kate"
+...(kate@localhost)2>   end
+...(kate@localhost)2> end
+```
+
+#### Sending Messages
+
+Now, we can use `Node.spawn_link/2` to have the `alex` node ask the `kate` node to call the `say_name` function:
+
+```elixir
+iex(alex@localhost)8> Node.spawn_link(:kate@localhost, fn -> Kate.say_name end)
+Hi, my name is Kate
+#PID<10507.132.0>
+```
+
+#### A Note on I/O and Nodes
+
+Notice that, although `Kate.say_name` is getting executed on the remote node, it is the local, or calling, node that receives the `IO.puts` output. That is because the local node is the **group leader**. The Erlang VM manages I/O via processes. This allows us to execute I/O tasks, like `IO.puts`, across distributed nodes. These distributed processes are managed by the I/O process group leader. The group leader is always the node that spawns the process. So, since our `alex` node is the one from which we called `spawn_link`, that node is the group leader and the output of `IO.puts` will be directed to the standard output stream of that node.
+
+#### Responding to Messages
+
+What if we want the node that receives the message to send some *response* back to the sender? We can use a simple `receive` and [`send`](https://hexdocs.pm/elixir/Process.html#send/3) setup to accomplish exactly that.
+
+We'll have our `alex` node spawn a link to the `kate` node and give the `kate` node an anonymous function to execute. That anonymous function will listen for the receipt of a particular tuple describing a message and the PID of the `alex` node. It will respond to that message by `send`-ing back a message to the PID of the `alex` node:
+
+```elixir
+iex(alex@localhost)2> pid = Node.spawn_link :kate@localhost, fn ->
+...(alex@localhost)2>   receive do
+...(alex@localhost)2>     {:hi, alex_node_pid} -> send alex_node_pid, :sup?
+...(alex@localhost)2>   end
+...(alex@localhost)2> end
+#PID<10467.112.0>
+iex(alex@localhost)3> pid
+#PID<10467.112.0>
+iex(alex@localhost)4> send(pid, {:hi, self()})
+{:hi, #PID<0.106.0>}
+iex(alex@localhost)5> flush()
+:sup?
+:ok
+```
+
+#### A Note On Communicating Between Nodes on Different Networks
+If you want to send messages between nodes on different networks, we need to start the named nodes with a shared cookie:
+
+
+```bash
+iex --sname alex@localhost --cookie secret_token
+```
+
+```bash
+iex --sname kate@localhost --cookie secret_token
+```
+
+Only nodes started with the same `cookie` will be able to successfully connect to one another.
+
+#### `Node.spawn_link` Limitations
```

Comment:
Function references should include arity.  This applies here and in the paragraphs belong.